data_structures/static_rtree.hpp

```diff
@@ -642,11 +642,56 @@ class StaticRTree
         return result_coordinate.is_valid();
     }
 
+    /**
+     * Checks whether A is between B-range and B+range, all modulo 360
+     * e.g. A = 5, B = 5, range = 10 == true
+     *      A = -6, B = 5, range = 10 == false
+     *      A = -2, B = 355, range = 10 == true
+     *      A = 6, B = 355, range = 10 == false
+     *      A = 355, B = -2, range = 10 == true
+     *
+     * @param A the bearing to check, in degrees, 0-359, 0=north
+     * @param B the bearing to check against, in degrees, 0-359, 0=north
+     * @param range the number of degrees either side of B that A will still match
+     * @return true if B-range <= A <= B+range, modulo 360
+     * */
+    static bool IsBearingWithinBounds(const int A,
+                                      const int B,
+                                      const int range)
```

Comment:
Right now `range` is known at compile-time. Will this ever change?

If not tell the compiler this useful information, such as:

    template <int Range>
    fn() {
        static_assert(range >= 0,"");
        //
    }

Also: if the range is never allowed to be negative, then either assert this everywhere or use an integral unsigned type.